### PR TITLE
Raise exception if fitness is None after calls to objectives

### LINKED ISSUE #309: Raise exception if any individual has fitness `None` after calling objectives
Without this check, difficult-to-interpret error messages can be generated in subsequent code.

### PR DESCRIPTION
**Pull request by jakobj**
as the title suggests. this should help to pinpoint bugs in objective functions.

closes #309

### CODE REVIEW

File: test/test_ea_mu_plus_lambda.py [hunk 1]
```diff
@@ -337,3 +337,15 @@ def test_mutate(population_params, genome_params, ea_params):
     assert np.any(
         [off_orig != off_mutated for off_orig, off_mutated in zip(offspring_original, offspring)]
     )
+
+
+def test_objective_must_set_valid_fitness(population_params, genome_params, ea_params):
+    def objective(ind):
+        return ind
```
Comment (HenrikMettler) [inline test/test_ea_mu_plus_lambda.py:344]: Maybe add `# Missing fitness assignement`

**Review by HenrikMettler — CHANGES_REQUESTED** [review]
Just a small suggestion to make the test easier to understand

File: test/test_ea_mu_plus_lambda.py [hunk 2]
```diff
@@ -337,3 +337,15 @@ def test_mutate(population_params, genome_params, ea_params):
     assert np.any(
         [off_orig != off_mutated for off_orig, off_mutated in zip(offspring_original, offspring)]
     )
+
+
+def test_objective_must_set_valid_fitness(population_params, genome_params, ea_params):
+    def objective(ind):
```
Comment (HenrikMettler) [inline test/test_ea_mu_plus_lambda.py:343]:
```suggestion
def test_objective_must_set_valid_fitness(population_params, genome_params, ea_params):
    def objective(ind):
        # missing ind.fitness assignement
```

**Comment by jakobj** [comment]
CI is happy, if you're happy, please approve this PR @HenrikMettler so we can merge it

**Comment by jakobj** [comment]
ping @HenrikMettler

**Review by HenrikMettler — APPROVED** [review]
LGTM, not sure who needs to squish the commits

**Comment by HenrikMettler** [comment]
> LGTM, not sure who needs to squish the commits

ping @jakobj

**Comment by jakobj** [comment]
squashed and pushed. let's see whether CI is happy :)

**Pull request merged**